sql/core/src/main/scala/org/apache/spark/sql/execution/python/FlatMapGroupsInPandasExec.scala

```diff
@@ -52,16 +58,27 @@ case class FlatMapGroupsInPandasExec(
     val bufferSize = inputRDD.conf.getInt("spark.buffer.size", 65536)
     val reuseWorker = inputRDD.conf.getBoolean("spark.python.worker.reuse", defaultValue = true)
     val chainedFunc = Seq(ChainedPythonFunctions(Seq(pandasFunction)))
-    val argOffsets = Array((0 until child.schema.length).toArray)
+    val argOffsets = Array((0 until (child.output.length - groupingAttributes.length)).toArray)
+    val schema = StructType(child.schema.drop(groupingAttributes.length))
 
     inputRDD.mapPartitionsInternal { iter =>
-      val grouped = GroupedIterator(iter, groupingAttributes, child.output)
+      val grouped = if (groupingAttributes.isEmpty) {
+        Iterator(iter)
```

Comment:
I don't totally understand empty groupby, could you please explain a bit to me?

What's the semantics of empty group? Does it pass all data in the partition as one group? Does df.groupby() shuffle all data into a single partition?

Reply:
The semantics of empty group is the latter. `df.groupby()` shuffle all data into a single partition.

Reply:
Gotcha.